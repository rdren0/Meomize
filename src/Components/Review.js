import React, { Component } from 'react';
import Card from './Card';
// import {cards} from './data';


class Review extends Component {
  constructor() {
    super();
    this.state = {
      filterCards:[]
      };
  }

  selectCategory = (e) => {
    let clickedButton = e.target.value;
    let filterCards = this.props.cards.filter(card => {
      return card.type.includes(clickedButton);
    })
    this.setState({
      filterCards
    })
  }

  showAll = (e) => {
    let filterCards = this.props.cards.map(card => {
      return card;
    })
    this.setState({
      filterCards
    })
  }



  render() {
      const flashCards = this.state.filterCards.map(card => {
      return <Card 
                key={card.id} 
                cardType={card.type} 
                cardTitle={card.title} 
                cardAnswer={card.answer} /> 
        })


    return (
      <div className="card-display">
      <h4 className="categories-text"> Sort by Category: </h4>
        <button value="return" className="filter-buttons" onClick={this.selectCategory}>RETURNS</button> 
        <button value="syntax" className="filter-buttons" onClick={this.selectCategory}>SYNTAX</button> 
        <button value="useCase" className="filter-buttons" onClick={this.selectCategory}>USE CASE</button>
        <button className="filter-buttons" onClick={this.showAll}>ALL</button>
        <div className='card-container'>
          {flashCards}
        </div>
      </div>
    );
  }

  
}


export default Review;



      