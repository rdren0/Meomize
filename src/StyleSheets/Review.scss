.card-container{
  background-color: #282c34;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}