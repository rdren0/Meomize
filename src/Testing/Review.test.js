import React from 'react';
import Review from '../Components/Review';
import { shallow } from 'enzyme';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

const selectCategory = jest.fn();

const mockCard = {
"id": 2,
"type": "syntax",
"title": "What goes in the parameters of a Reduce?",
"answer": "Acc and Current Value",
"complete": false
}

describe('Review', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Review
                key={mockCard.id} 
                cardType={mockCard.type} 
                cardTitle={mockCard.title} 
                cardAnswer={mockCard.answer} />
    )
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('should have the proper defualt state', () => {
    expect(wrapper.state('complete')).toEqual(false);
  });



});