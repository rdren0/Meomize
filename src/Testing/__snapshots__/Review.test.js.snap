// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should match the snapshot with all data passed in correctly 1`] = `
<div
  className="card-display"
>
  <h4
    className="categories-text"
  >
     Sort by Category: 
  </h4>
  <button
    className="filter-buttons"
    onClick={[Function]}
    value="return"
  >
    RETURNS
  </button>
  <button
    className="filter-buttons"
    id="syntax-button"
    onClick={[Function]}
    value="syntax"
  >
    SYNTAX
  </button>
  <button
    className="filter-buttons"
    onClick={[Function]}
    value="useCase"
  >
    USE CASE
  </button>
  <button
    className="filter-buttons"
    onClick={[Function]}
  >
    ALL
  </button>
  <div
    className="card-container"
  />
</div>
`;
